This is a backport of this fix to version 2.1.3:
https://github.com/gcanti/io-ts/pull/657

It allows Typescript to be upgraded beyond version 4.7.4

diff --git a/node_modules/io-ts/lib/index.d.ts b/node_modules/io-ts/lib/index.d.ts
index 17c4ce8..1a7545c 100644
--- a/node_modules/io-ts/lib/index.d.ts
+++ b/node_modules/io-ts/lib/index.d.ts
@@ -996,6 +996,6 @@ export declare type Exact<T, X extends T> = T & {
  * @since 1.1.0
  * @deprecated
  */
-export declare function alias<A, O, P, I>(codec: PartialType<P, A, O, I>): <AA extends Exact<A, AA>, OO extends Exact<O, OO> = O, PP extends Exact<P, PP> = P, II extends I = I>() => PartialType<PP, AA, OO, II>;
-export declare function alias<A, O, P, I>(codec: StrictType<P, A, O, I>): <AA extends Exact<A, AA>, OO extends Exact<O, OO> = O, PP extends Exact<P, PP> = P, II extends I = I>() => StrictType<PP, AA, OO, II>;
-export declare function alias<A, O, P, I>(codec: InterfaceType<P, A, O, I>): <AA extends Exact<A, AA>, OO extends Exact<O, OO> = O, PP extends Exact<P, PP> = P, II extends I = I>() => InterfaceType<PP, AA, OO, II>;
+export declare function alias<A, O, P, I>(codec: PartialType<P, A, O, I>): <AA extends A, OO extends O = O, PP extends P = P, II extends I = I>() => PartialType<PP, AA, OO, II>;
+export declare function alias<A, O, P, I>(codec: StrictType<P, A, O, I>): <AA extends A, OO extends O = O, PP extends P = P, II extends I = I>() => StrictType<PP, AA, OO, II>;
+export declare function alias<A, O, P, I>(codec: InterfaceType<P, A, O, I>): <AA extends A, OO extends O = O, PP extends P = P, II extends I = I>() => InterfaceType<PP, AA, OO, II>;
